dspace-server-webapp/src/main/java/org/dspace/app/rest/AltchaCaptchaRestController.java

```diff
@@ -0,0 +1,181 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.random.RandomGenerator;
+
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.apache.commons.codec.digest.HmacUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.tomcat.util.http.FastHttpDateFormat;
+import org.dspace.services.ConfigurationService;
+import org.json.JSONObject;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.rest.webmvc.ControllerUtils;
+import org.springframework.hateoas.Link;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * ALTCHA (Proof of Work, cookie-less) controller to handle challenge requests.
+ * A salt, challenge hash will be sent and the client will have to calculate the number and send it back
+ * to implementing controllers (e.g. request-a-copy) for validation.
+ * The proof-of-work makes spam uneconomic, without requiring annoying puzzle tests or 3rd party services.
+ * @see org.dspace.app.rest.repository.RequestItemRepository
+ * @see <a href="https://altcha.org/docs/server-integration">Altcha docs></a>
+ *
+ * @author Kim Shepherd
+ */
+@RequestMapping(value = "/api/captcha")
```

Comment:
We need to document these new endpoints in our REST Contract https://github.com/DSpace/RestContract/blob/main/README.md